giflens/GifLens
# Side menu

Ticket: https://github.com/giflens/GifLens/projects/1#card-22245372 <br>
Issue: fixes #19, fixes #27 

## Changes

- added a side menu (view Container)
- two views are accessible, history and favorites (only history is concerned by this PR)
- searching a gif and selecting adds it to the history
- clicking on the link in the history adds the gif as a GIFLENS tag ( temporary, might be replace by an icon action only)

## Remaining [WIP]

- [X] Limit to 15 the number of entries in the history 
- [x] Allow to delete an entry
- [x] Refactor search to have the add Gif action factorized
- [x] Adapt old search tests
- [x] Create test for all new features

## How To Test

- Run the extension, and search a GIF
- confirm that the title of the GIF appears in the history once one is selected

## Applicable Research Resources

- https://code.visualstudio.com/api/extension-guides/tree-view
- https://developers.giphy.com/docs/
- https://github.blog/2013-05-14-closing-issues-via-pull-requests/
